tests/compile-fail/while_loop.rs

```diff
@@ -30,12 +22,13 @@ fn main() {
             None => break
         };
     }
-    loop { // no error, match is not the only statement
-        match y {
-            Some(_x) => true,
+    loop { //~ERROR
+        let z = match y {
+            Some(x) => x,
             None => break
         };
-        let _x = 1;
```

Comment:
@llogiq that could still be `while...let`, as told in #262 which is the inspiration for this PR :)